xbmc/guilib/GUIEditControl.cpp

```diff
@@ -743,7 +743,7 @@ void CGUIEditControl::ValidateInput()
 
 void CGUIEditControl::SetFocus(bool focus)
 {
-  m_smsTimer.Stop();
+  UpdateText();
```

Comment:
there is no need to send an update (validate input +  send click message etc.) in `UpdateText()`. You should use `UpdateText(false)` which only invalidate the control and stops m_smsTimer.